app/models/concerns/votable.rb

```diff
@@ -0,0 +1,48 @@
+module Votable
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :attachments, as: :attachmentable, dependent: :destroy
+    has_many :votes, as: :votable, dependent: :destroy
+    belongs_to :user
+
+    accepts_nested_attributes_for :attachments, reject_if: :all_blank, allow_destroy: true
+    accepts_nested_attributes_for :votes
+  end
+
+  def liked_by(user)
+    vote = votes.where(user_id: user.id).first
+    if vote
+      update_vote_for(vote) if vote.choice == false
```

Comment:
`if !vote.choice`

And it's better to use numeric values for votes (i.e. "+1" and "-1") for several reasons:
- easier to calculate total rating (just sum of values);
- easier to extend voting logic and add, for example, weighted votes (consider a possible feature "Vote of a user with a high reputation costs 2x");
- ... or implement scaled voting ("bad", "normal", "good", "awesome", etc).